stl/inc/format

```diff
@@ -0,0 +1,363 @@
+// format standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _FORMAT_
+#define _FORMAT_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#if !_HAS_CXX20
+#pragma message("The contents of <format> are available only with C++20 or later.")
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+
+#include <cassert>
+#include <charconv>
+#include <concepts>
+#include <exception>
+#include <iterator>
+#include <locale>
+#include <string>
+#include <string_view>
+#include <variant>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+
+
+class format_error : public _STD runtime_error {
+    using _STD runtime_error::runtime_error;
+};
+
+enum class _Align { _None, _Left, _Right, _Center };
+
+struct _Auto_id_tag {};
+
+// clang-format off
+template <typename _Ty, typename _CharT>
+concept _Parse_spec_callbacks = requires(_Ty _At, _STD basic_string_view<_CharT> _Sv, _Align _Aln) {
+    { _At._On_align(_Aln) } -> same_as<void>;
+    { _At._On_fill(_Sv) } -> same_as<void>;
+    { _At._On_width(_STD declval<int>()) } -> same_as<void>;
+    { _At._On_dynamic_width(_STD declval<int>()) } -> same_as<void>;
+    { _At._On_dynamic_width( _STD declval<_Auto_id_tag>()) } -> same_as<void>;
+};
+template<typename _Ty, typename _CharT>
+concept _Parse_arg_id_callbacks = requires(_Ty _At) {
+    { _At._On_auto_id() } -> same_as<void>;
+    { _At._On_manual_id(_STD declval<int>()) } -> same_as<void>;
+};
+//clang-format on
+
+template<typename _CharT>
+const _CharT* _Parse_nonnegative_integer(const _CharT* _Begin, const _CharT* _End, int& _Integer) {

```

Comment:
the main reason is that this is what libfmt does (because it works on compilers that don't have string view, and also the functions tend to manipulate the pointers directly, I'm tempted to change this stuff to string_views, or some other sort of range.

Reply:
I tried to change it to string views, it didn't work that well, perhaps passing around "view, iterator (current parsing position)" pairs would work better, but I'm not sure it's worth it, this keeps things closer to libfmt.